addons/survey/models/survey_survey.py

```diff
@@ -435,3 +489,40 @@ def action_survey_user_input_invite(self):
                     'search_default_invite': 1})
         action['context'] = ctx
         return action
+
+    @api.multi
+    def has_attempts_left(self, partner, email):
+        self.ensure_one()
+
+        if self.access_mode != 'public' and self.limit_number_of_attempts:
+            return self.attempts_left(partner, email) > 0
+
+        return True
+
+    @api.multi
+    def attempts_left(self, partner, email):
+        self.ensure_one()
```

Comment:
Maybe a docstring explaining this method return the number of attempts left (and not a boolean as i thought) is usefull here. Or rename the method. ;)

Reply:
Added a docstring. Method name makes sense as discussed.